tls/s2n_handshake_io.c

```diff
@@ -1608,6 +1608,11 @@ int s2n_negotiate_impl(struct s2n_connection *conn, s2n_blocked_status *blocked)
     POSIX_ENSURE_REF(conn);
     POSIX_ENSURE_REF(blocked);
 
+    /* SSLv3 handshake is not supported when built with openssl-fips */
+    if (s2n_is_in_fips_mode() && conn->client_protocol_version == S2N_SSLv3 && !s2n_libcrypto_is_awslc()) {
```

Comment:
This is probably not the correct place for this logic. I recommend putting this check at the place where we decide the protocol version, likely somewhere in the Client Hello method.